src/test/java/spoon/reflect/ast/CloneTest.java

```diff
@@ -265,4 +269,53 @@ public void testIssue3389() {
 		// contract: clone preserves the order
 		assertEquals(c.getFields().get(0), c.clone().getTypeMembers().get(0));
 	}
+
+	@Test
+	public void testCloneKeepsImplicitModifierState() {
+		// contract: When cloning an element the implicit state is kept as well
+		CtClass<?> ctClass = Launcher.parseClass(
+				"class Foo { interface Nested { void bar(); } }"
+		);
+		CtMethod<?> method = ctClass.getNestedType("Nested").getMethod("bar");
+
+		Set<CtExtendedModifier> expectedModifiers = new HashSet<>();
+		expectedModifiers.add(new CtExtendedModifier(ModifierKind.PUBLIC, true));
+		expectedModifiers.add(new CtExtendedModifier(ModifierKind.ABSTRACT, true));
+
+		assertEquals(
+				expectedModifiers,
+				method.getExtendedModifiers()
+		);
+
+		assertEquals(
+				expectedModifiers,
+				method.clone().getExtendedModifiers()
+		);
```

Comment:
Same thing here, this might be rather fragile if extended modifiers are altered. Could you make this test more explicit in checking implicitness in particular?

I'd personally not parse this from source code, but instead just create anything modifiable, explicitly add modifiers to it, and then check that for the given modifiers the implicitness doesn't change when cloning. That is to say, don't use equals of the modifier object, but explicitly check the value of `isImplicit()` for some given modifier kind.